### PR TITLE
Verbose query execution time

### PR DESCRIPTION
**Pull request by jmatak**
Useful information included from #35 

- [x] Check formatter

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -59,6 +59,9 @@ DEFINE_bool(term_colors, false, "Use terminal colors syntax highlighting.");
 DEFINE_string(output_format, "tabular",
               "Query output format. Can be csv/tabular. If output format is "
               "other than tabular `fit-to-screen` flag is ignored.");
+DEFINE_bool(verbose_query, false,
```
Comment (gitbuda) [inline src/main.cpp:62]: This was just an example at hand in the #35, but maybe a better flag name would be `--print-execution-time-info` or something similar which is aligned with the actual change, anyway, something to think about and maybe adjust 😄

File: src/utils/utils.cpp [hunk 2]
```diff
@@ -601,6 +604,21 @@ void EchoInfo(const std::string &message) {
   }
 }
 
+void EchoExecutionTimeInfo(const std::map<std::string, double> &execution_info) {
+   std::cout << "Additional execution time info:" << std::endl;
+  for (const auto &[key, value] : execution_info) {
+    if (key == "cost_estimate") {
```
Comment (gitbuda) [inline src/utils/utils.cpp:610]: `cost_estimate` is not a time measure -> `--verbose-query` makes more sense

File: src/utils/utils.cpp [hunk 3]
```diff
@@ -601,6 +604,21 @@ void EchoInfo(const std::string &message) {
...
+      std::cout << "Cost estimate: " << value << std::endl;
+    }else if (key == "parsing_time") {
+      std::cout << "Query parsing time: " << value << " sec" << std::endl;
+    }else if (key == "planning_time") {
+      std::cout << "Query planning time: " << value << " sec" <<  std::endl;
+    }else if (key == "plan_execution_time") {
+      std::cout << "Query plan execution time: " << value << " sec" << std::endl;
```
Comment (gitbuda) [inline src/utils/utils.cpp:617]: Formatting here seems wrong

File: src/utils/utils.cpp [hunk 4]
```diff
@@ -850,6 +864,19 @@ QueryData ExecuteQuery(mg_session *session, const std::string &query) {
...
+      std::map<std::string, double> execution_time_info;
+      execution_time_info.emplace(
+          "cost_estimate", ParseFloat(mg_map_at(summary, "cost_estimate")));
+      execution_time_info.emplace(
+          "parsing_time", ParseFloat(mg_map_at(summary, "parsing_time")));
+      execution_time_info.emplace(
+          "planning_time", ParseFloat(mg_map_at(summary, "planning_time")));
+      execution_time_info.emplace(
+          "plan_execution_time",
+          ParseFloat(mg_map_at(summary, "plan_execution_time")));
```
Comment (gitbuda) [inline src/utils/utils.cpp:877]: This can be done in a `for` loop 😄

File: src/utils/utils.cpp (hunk 3 above)
Comment (jmatak) [reply src/utils/utils.cpp:617]:
```
memgraph> UNWIND [1, 2] AS number RETURN number;
+--------+
| number |
+--------+
| 1      |
| 2      |
+--------+
2 rows in set (round trip in 0.001 sec)
Additional execution time info:
Cost estimate: 1.3
Query parsing time: 0.0201984 sec
Query plan execution time: 5.4838e-05 sec
Query planning time: 0.000699963 sec
```

This is the example formatting

File: src/utils/utils.cpp (hunk 2 above)
Comment (jmatak) [reply src/utils/utils.cpp:610]: Because of time estimate we are going with `-verbose_query`?

File: src/main.cpp [hunk 5]
```diff
@@ -274,12 +252,10 @@ int main(int argc, char **argv) {
       console::EchoInfo("Bye");
       break;
     }
-    if (query->empty())
-      continue;
+    if (query->empty()) continue;
```
Comment (jbajic) [inline src/main.cpp:255]:
```suggestion
    if (query->empty()) {
      continue;
    }
```

File: src/main.cpp [hunk 6]
```diff
@@ -274,12 +252,10 @@ int main(int argc, char **argv) {
...
     try {
       auto ret = query::ExecuteQuery(session.get(), *query);
-      if (ret.records.size() > 0)
-        Output(ret.header, ret.records, output_opts, csv_opts);
+      if (ret.records.size() > 0) Output(ret.header, ret.records, output_opts, csv_opts);
```
Comment (jbajic) [inline src/main.cpp:258]: I would fix this on every occurrence of format changes
```suggestion
      if (ret.records.size() > 0) {
        Output(ret.header, ret.records, output_opts, csv_opts);
      }
```

File: src/main.cpp [hunk 7]
```diff
@@ -301,6 +277,9 @@ int main(int argc, char **argv) {
         if (ret.stats) {
           console::EchoStats(ret.stats.value());
         }
+        if (FLAGS_verbose_query) {
+          console::EchoExecutionTimeInfo(ret.execution_time_info.value());
+        }
```
Comment (jbajic) [inline src/main.cpp:282]: Also, check if there is `exeuction_time_info`, some queries do not have it (e.g. `CREATE INDEX ON :label(prop);`)

**Review by jbajic — CHANGES_REQUESTED** [review]
Also please add tests, and bump up the `mgconsole` version

File: src/main.cpp (hunk 5 above)
Comment (jmatak) [reply src/main.cpp:255]: Fixed in clang PR: #37

File: src/main.cpp (hunk 6 above)
Comment (jmatak) [reply src/main.cpp:258]: Fixed in clang PR: #37

File: src/main.cpp (hunk 7 above)
Comment (jmatak) [reply src/main.cpp]: Covered

**Comment by jmatak** [comment]
@jbajic Currently, no tests check additional flags and the full output. Do you think there is a benefit to adding these? Also, how should we test in end-to-end something that is not deterministic like execution time

**Review by jbajic — APPROVED** [review]
Nice!

**Comment by jbajic** [comment]
You are right, feel free to ignore the tests, but please bump up `mgconsole` version in CMake!

> @jbajic Currently, no tests check additional flags and the full output. Do you think there is a benefit to adding these? Also, how should we test in end-to-end something that is not deterministic like execution time

**Pull request merged**